x-pack/plugins/stack_connectors/public/connector_types/security/tines/tines_params.tsx

```diff
@@ -0,0 +1,330 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import {
+  EuiBadge,
+  EuiCallOut,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiHighlight,
+  EuiSpacer,
+} from '@elastic/eui';
+import type { ActionParamsProps } from '@kbn/triggers-actions-ui-plugin/public/types';
+import {
+  JsonEditorWithMessageVariables,
+  useSubAction,
+  useKibana,
+} from '@kbn/triggers-actions-ui-plugin/public';
+import { SUB_ACTION } from '../../../../common/connector_types/security/tines/constants';
+import type {
+  TinesStoryObject,
+  TinesWebhookObject,
+  TinesWebhooksActionParams,
+  TinesStoriesActionResponse,
+  TinesWebhooksActionResponse,
+  TinesStoriesActionParams,
+} from '../../../../common/connector_types/security/tines/types';
+import type { TinesExecuteActionParams, TinesExecuteSubActionParams } from './types';
+import * as i18n from './translations';
+
+type StoryOption = EuiComboBoxOptionOption<TinesStoryObject>;
+type WebhookOption = EuiComboBoxOptionOption<TinesWebhookObject>;
+
+const createOption = <T extends TinesStoryObject | TinesWebhookObject>(
+  item: T
+): EuiComboBoxOptionOption<T> => ({
+  key: item.id.toString(),
+  value: item,
+  label: item.name,
+});
+
+const renderStory = (
+  { label, value }: StoryOption,
+  searchValue: string,
+  contentClassName: string
+) => (
+  <EuiFlexGroup className={contentClassName} direction="row" alignItems="center">
+    <EuiFlexItem grow={false}>
+      <EuiHighlight search={searchValue}>{label}</EuiHighlight>
+    </EuiFlexItem>
+    {value?.published && (
+      <EuiFlexItem grow={false}>
+        <EuiBadge color="hollow">{i18n.STORY_PUBLISHED_BADGE_TEXT}</EuiBadge>
```

Comment:
The `Published` badge was requested by Tines:

![story_published_badge](https://user-images.githubusercontent.com/17747913/200539511-1e29fb9f-a278-46b1-b846-9c809402dca4.png)